### PR TITLE
Add DlrNDArray & DlrNDManager to make DLR work alone

### PR DESCRIPTION
**Pull request by stu1130**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrNDManager.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        super(parent, device);
+    }
+
+    static DlrNDManager getSystemManager() {
+        return SYSTEM_MANAGER;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public NDArray create(float[] data, Shape shape) {
```
Comment (lanking520) [inline dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrNDManager.java:38]: Can we create from Buffer? This seemed to be the most common way people would do. Also probably you need to override the other creation method to avoid calling from it

Comment (stu1130) [reply dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrNDManager.java:38]: I was thinking about that as well. After I searched [Buffer vs primitive array](https://stackoverflow.com/questions/18913001/when-to-use-array-buffer-or-direct-buffer). array seem to be a better choice cause we eventually will pass float[] to JNI not the data pointer or byte array

Comment (frankfliu) [reply dlr/dlr-engine/src/main/java/ai/djl/dlr/engine/DlrNDManager.java:38]: I think we should use Buffer, if it's DirectBuffer, we don't even need copy in JNI.

**Review by frankfliu — APPROVED** [review]
